packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +189,176 @@ class Tooltip extends mixin(
     this.tooltip.changeState(
       state,
       Object.assign(detail, { delegatorNode: this.element }),
-      callback
+      () => {
+        // @questions
+        // -- What is the expected behavior if the user clicks on the trigger when the tooltip is already open
+        // -- What's the focus state look like when there is only rich text within a tooltip (no buttons/links/etc)
+        // -- There is a "flash" where the next element (another trigger in the demo) is focused before focus is
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        const isHidden = state !== 'shown';
+        this.tooltip.element.setAttribute('aria-hidden', isHidden.toString());
+        this.element.setAttribute('aria-expanded', (!isHidden).toString());
+
+        if (this._handleFocusinListener) {
+          this._handleFocusinListener = this.unmanage(
+            this._handleFocusinListener
+          ).release();
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
+          this._hookCloseActions(this.element);
+          const focusableNode = this.tooltip.element.querySelector(
+            settings.selectorTabbable
+          );
+          if (focusableNode) {
+            focusableNode.focus();
+          } else {
+            this.tooltip.element.focus();
+            const tooltipTabindex = this.tooltip.element.getAttribute(
+              'tabindex'
+            );
+            if (__DEV__ && [null, '-1'].includes(tooltipTabindex)) {
+              warning(
+                tooltipTabindex,
```

Comment:
The 1st argument of `warning()` is similar to `assert()` in C - That said, would you want to put `tooltipTabindex && tooltipTabindex !== '-1'` here? Another thing to note here is that `Array.prototype.includes()` is not in [our required polyfills list](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/demo/polyfills/index.js) and thus that new API cannot be used in the codebase.